src/MultiRegion/multi_region_cubed_sphere_grid.jl

```diff
@@ -424,11 +424,11 @@ function ConformalCubedSphereGrid(filepath::AbstractString, arch::AbstractArchit
     return MultiRegionGrid{FT, panel_topology[1], panel_topology[2], panel_topology[3]}(arch, partition, connectivity, region_grids, devices)
 end
 
-function with_halo(new_halo, csg::ConformalCubedSphereGrid) 
+function with_halo(new_halo, csg::ConformalCubedSphereGrid)
     region_rotation = []
 
-    for r in 1:length(csg.partition)
-        push!(region_rotation, rotation_from_panel_index(panel_index(r, csg.partition)))
+    for region in 1:length(csg.partition)
+        push!(region_rotation, csg[region].conformal_mapping.rotation)
```

Comment:
Is this the key change? What is the issue?

Reply:
The function rotation_from_panel_index is undefined within the code, leading to the error message: "ERROR: LoadError: UndefVarError: rotation_from_panel_index not defined". It's likely that the function's definition was removed in a previous commit. Hence this amendment suggested by @navidcy.

Reply:
Thanks @siddharthabishnu

Reply:
multi-region syntax prefers the definition of a function and the use `@apply_regionally`. We do not want to use loops because of the possibility of data being on different devices

Reply:
@simone-silvestri can you be more explicit about the change that is needed?